sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuAggregateExec.scala

```diff
@@ -277,25 +279,32 @@ object AggregateUtils extends Logging {
 
       val newBuckets = batchesByBucket.flatMap(bucket => {
         if (needRepartitionAgain(bucket)) {
-          if (hashSeed + 7 > 200) {
-            log.warn("Too many times of repartition, may hit a bug? Size for each batch in " +
+          if (recursiveDepth >= maxRecursiveDepth) {
+            // Normally this should not happen, because we are repartitioning data that has
+            // already went through first round of aggregation, so there shouldn't be too many
+            // duplicated rows (the duplication only happens in different batches) to prevent
+            // repartitioning out (considering we're changing seed each time we repartition).
+            // However for some test cases with really small batch size, this can happen. So
+            // we're just logging some warnings here.
+            log.warn("The bucket is still too large after " + recursiveDepth +
+              " times of repartition, may hit a bug? Size for each batch in " +
               "current bucket: " + bucket.map(_.sizeInBytes).mkString(", ") + " rows: " +
               bucket.map(_.numRows()).mkString(", ") + " targetMergeBatchSize: "
               + targetMergeBatchSize)
```

Comment:
What bug? 
What are the consequences? 
What action can the user take after seeing such a scary warning?

Reply:
hi @gerashegalov , are you suggesting me to handle it like in https://github.com/NVIDIA/spark-rapids/blob/346b76310e6e68e24665669999882dc97694ba24/sql-plugin/src/main/scala/com/nvidia/spark/rapids/window/GpuWindowExpression.scala#L332 ?

Reply:
hi @gerashegalov ,  please check https://github.com/NVIDIA/spark-rapids/pull/11792#discussion_r1874074147, it's not a bug, and with the followup issue https://github.com/NVIDIA/spark-rapids/issues/11834 opened by Bobby, do you think I can resolve this thread?

Reply:
Why  is this warning still saying "may hit a bug?" then?

Reply:
hi @gerashegalov , the message is refined, please check